src/System.Linq/src/System/Linq/In.cs

```diff
@@ -0,0 +1,38 @@
+﻿//NiceLinq by Mohamed Elshawaf m_shawaf@outlook.com
+using System.Collections.Generic;
+using System.Diagnostics;
+using static System.Linq.Utilities;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
```

Comment:
There's no `Queryable` equivalent, so trying this on another linq provider will always take everything from the source into memory. Right now one can do `source.Where(m => values.Contains(identifier(m)))` on linq2sql or EF and have that run on the database.